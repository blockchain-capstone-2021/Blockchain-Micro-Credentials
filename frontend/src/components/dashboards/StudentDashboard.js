import React, { useState, useEffect } from 'react'
import { Link, useHistory } from 'react-router-dom';
import microcredapi from '../../apis/microcredapi'
import "./Dashboard.css";


const StudentDashboard = (props) => {
  const history = useHistory()

  const [availableEnrolments, setAvailableEnrolments] = useState()
  const [unavailableEnrolments, setUnavailableEnrolments] = useState()
  const [unitMap, setUnitMap] = useState()
  useEffect(() => {
    async function initState(){
      await microcredapi.get(`/student/${window.localStorage.getItem('userId')}/enrolled`).then(response => {
        setUnavailableEnrolments(response.data.enrolments.unavailable.length > 0 ? response.data.enrolments.unavailable: undefined)
        setAvailableEnrolments(response.data.enrolments.available.length > 0 ? response.data.enrolments.available: undefined)
        setUnitMap(response.data.unitMap)
      })
    }
    initState()
  }, [])

  function renderEnrolments(enrolmentsArray, type) {
    const isAvailable = type == 'AVAILABLE'? true : false
    return enrolmentsArray.map(enrolment => {
      return (
        <div className={`card`} style={{width: '18rem'}}>
        <div className="card-body">
          <h5 className="card-title">{unitMap ? unitMap[enrolment.unitId]: "Loading"}</h5>
          <p className="card-text">Enrolled for:<br/>{enrolment.semOfEnrolment}</p>
          {
            isAvailable?
            <Link to={`/unit/${enrolment.unitId}`} className="btn btn-primary" onClick={() => {window.localStorage.setItem('enrolmentPeriod', enrolment.semOfEnrolment)}}>Go</Link>:
            ""
          }
        </div>
      </div>
      )
    })
  }

  return (
    <div className="jumbotron align-center">
    <main role="main" >
      <div className="d-flex justify-content-between flex-wrap flex-md-nowrap align-items-center pt-3 pb-2 mb-3 border-bottom">
        <h1 className="h2">Dashboard</h1>
      </div>
      <section className="mb-5">
      <h2 className="mt-2">Enrolments</h2>
      <h3 className="py-5">Available</h3>
      {availableEnrolments ? renderEnrolments(availableEnrolments, 'AVAILABLE') :  "No available enrolments"}
      <h3 className="py-5">Completed</h3>
      {unavailableEnrolments ? renderEnrolments(unavailableEnrolments, 'UNAVAILABLE') : "No unavailable enrolments"}
      </section>
    </main>
  </div>
  )
}

export default StudentDashboard
