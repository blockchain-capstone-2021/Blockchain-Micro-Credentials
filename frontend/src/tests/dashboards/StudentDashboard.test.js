import React from "react";
import StudentDashboard from "../../components/dashboards/StudentDashboard";
import {shallow} from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

const wrapper = shallow(<StudentDashboard />);

describe('testing Student Dashboard component', () => {
    it('should be defined on render', () => {
        expect(StudentDashboard).toBeDefined();
        expect(wrapper).toMatchSnapshot();
        // expect(axios.get).toHaveBeenCalledTimes(1);
    });
    it('renders a header', () => {
        expect(wrapper.find("h1")).toHaveLength(1);
    });
    it('renders two tables', () => {
        expect(wrapper.find("table")).toHaveLength(2);
    });
    it('renders message if no enrolments exist', () => {
        expect(
            wrapper.containsMatchingElement(
              <tbody>No available enrolments</tbody>
            )
          ).toBeTruthy();
    });
    // it('renders enrolments if enrolments exist', () => {
    //     await microcredapi.get(`/student/s3724266/enrolled`)
    //     expect(wrapper).toMatchSnapshot();
    // });
})