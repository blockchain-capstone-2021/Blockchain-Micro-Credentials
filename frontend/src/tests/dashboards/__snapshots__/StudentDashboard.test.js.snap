// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Student Dashboard component should be defined on render 1`] = `ShallowWrapper {}`;
